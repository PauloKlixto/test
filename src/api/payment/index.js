import request from '@/api/plugins/request'

export default {
  async fechPayment(body, success, error, done) {
    return await request(
      {
        method: 'post',
        url: 'pagar/',
        body
      },
      success,
      error,
      done
    )
  }
}
