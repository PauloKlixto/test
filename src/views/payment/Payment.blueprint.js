export default paymentOptions => [
  {
    id: 'card_number',
    tag: 'InputGroup',
    model: 'card_number',
    rules: ['required'],
    props: {
      mask: 'card',
      label: 'Número do cartão',
      type: 'text',
      floatingLabel: true,
      invalidLabel: 'Número de cartão inválido'
    },
    events: {
      input: 'getCardType'
    }
  },
  {
    id: 'card_name',
    tag: 'InputGroup',
    model: 'card_name',
    rules: ['required'],
    props: {
      label: 'Nome (igual ao cartão)',
      type: 'text',
      floatingLabel: true,
      invalidLabel: 'Insira seu nome completo'
    }
  },
  {
    id: 'card_date',
    tag: 'InputGroup',
    model: 'card_date',
    rules: ['required'],
    props: {
      mask: 'cardDate',
      label: 'Validade',
      type: 'text',
      floatingLabel: true,
      invalidLabel: 'Data inválida'
    }
  },
  {
    id: 'card_cvv',
    tag: 'InputGroup',
    model: 'card_cvv',
    rules: ['required'],
    props: {
      label: 'CVV',
      type: 'number',
      floatingLabel: true,
      invalidLabel: 'Código inválido'
    }
  },
  {
    id: 'card_installments',
    tag: 'BaseSelect',
    model: 'card_installments',
    rules: ['required'],
    props: {
      label: 'Número de parcelas',
      options: paymentOptions,
      invalidLabel: 'insira o número de parcelas'
    }
  },
  {
    id: 'send',
    tag: 'BaseButton',
    props: {
      label: 'Continuar',
      type: 'submit'
    }
  }
]
