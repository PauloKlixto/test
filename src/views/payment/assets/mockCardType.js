export default {
  company: '',
  name: 'creditCardNumber',
  valid: false
}
