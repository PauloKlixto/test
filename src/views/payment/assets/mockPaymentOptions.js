export default [
  {
    id: 0,
    name: '12x R$1.000,00 sem juros'
  },
  {
    id: 1,
    name: '6x R$2.000,00 sem juros'
  },
  {
    id: 2,
    name: '3x R$4.000,00 sem juros'
  },
  {
    id: 3,
    name: '1x R$12.000,00 sem juros'
  }
]
