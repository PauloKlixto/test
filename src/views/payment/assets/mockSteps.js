export default [
  {
    id: 1,
    step: 1,
    name: 'Carrinho',
    active: true
  },
  {
    id: 2,
    step: 2,
    name: 'Pagamento',
    active: false
  },
  {
    id: 3,
    step: 3,
    name: 'Confirmação',
    active: false
  }
]
