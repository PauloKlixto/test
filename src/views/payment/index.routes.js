const Payment = () => import(/* webpackChunkName: "payment" */ './Payment.vue')

const routes = [
  {
    name: 'Payment',
    path: '/',
    component: Payment
  }
]

export default routes
