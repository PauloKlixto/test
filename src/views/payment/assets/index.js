import steps from './mockSteps'
import paymentOptions from './mockPaymentOptions'
import mockCardType from './mockCardType'

export { steps, paymentOptions, mockCardType }
