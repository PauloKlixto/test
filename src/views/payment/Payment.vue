<template>
  <main class="Payment">
    <section class="Payment__sidebar">
      <BaseBackButton
        class="Payment__backButton"
        label="Alterar forma de pagamento"
        is-inverted-theme
      />
      <BaseSteps :steps="steps" class="Payment__stepsInSidebar" invert-colors />
      <BaseHeading
        class="Payment__headingText"
        label="Adicione um novo cartão de crédito"
        icon="credit-card"
      />
      <BaseCreditCard
        class="Payment__creditCard"
        :card-number="form.card_number"
        :card-name="form.card_name"
        :card-date="form.card_date"
        :card-cvv="form.card_cvv"
        :company="cardType.company"
      />
    </section>
    <section class="Payment__form">
      <BaseSteps :steps="steps" class="Payment__steps" />
      <DynamicForm
        class="Payment__fields"
        :model="form"
        :blueprint="blueprint(paymentOptions)"
        :endpoint-errors="endpointErrors"
        @getCardType="getCardType"
        @submit="onSubmit"
      />
    </section>
  </main>
</template>

<script>
import {
  BaseBackButton,
  BaseCreditCard,
  BaseHeading,
  BaseSteps
} from '@/components/atoms'
import { DynamicForm } from '@/components/molecules'
import { steps, paymentOptions, mockCardType } from '@/views/payment/assets'
import blueprint from '@/views/payment/Payment.blueprint'
import Rules from '@doc88/flux-validator-js'

export default {
  name: 'Payment',
  components: {
    BaseBackButton,
    BaseCreditCard,
    BaseHeading,
    BaseSteps,
    DynamicForm
  },
  data() {
    return {
      form: {
        card_number: '',
        card_name: '',
        card_date: '',
        card_cvv: '',
        card_installments: ''
      },
      endpointErrors: {},
      cardType: mockCardType,
      steps,
      paymentOptions
    }
  },
  methods: {
    blueprint,

    getCardType() {
      this.cardType =
        this.form.card_number === ''
          ? mockCardType
          : Rules.creditCardNumber('creditCardNumber', this.form.card_number)
    },

    async onSubmit() {
      console.log('enviou')

      // const res = await payment.paymentData(this.form, (r) => r, (e) => e, () => this.setLoading(false))

      // if (res.validation) return this.endpointErrors = res.validation
      // if(!res.message) this.$router.push('/conclusion')
    }
  }
}
</script>

<style lang="scss" scoped>
$responsiveValue: 960px;

.Payment {
  background: $color-white;
  min-height: calc(100vh - 200px);
  display: grid;
  grid-template-columns: 352px 1fr;

  @media (max-width: $responsiveValue) {
    grid-template-columns: 1fr;
    grid-template-rows: 240px 1fr;
  }

  &__sidebar {
    background-color: $color-red;
    padding-top: $space-xl;
    padding-left: $space-xl;
    padding-bottom: $space-xl;
    padding-right: $space-md;

    @media (max-width: $responsiveValue) {
      display: grid;
      padding: 0;
      grid-template-columns: 40px 1fr 40px;
      grid-template-rows: 70px 70px 1fr;
      grid-template-areas: 'backButton stepsInSidebar .' '. headingText .' '. creditCard .';
      align-items: center;
      justify-items: center;
    }
  }

  &__backButton {
    margin-bottom: $space-xl;
    grid-area: backButton;
    @media (max-width: $responsiveValue) {
      margin-bottom: 0;
      margin-top: 45px;
    }
  }

  &__stepsInSidebar {
    display: none;
    @media (max-width: $responsiveValue) {
      display: block;
      grid-area: stepsInSidebar;
    }
  }

  &__headingText {
    margin-bottom: $space-lg;
    grid-area: headingText;
    max-width: 240px;
    height: 46px;
    @media (max-width: $responsiveValue) {
      margin-bottom: 0;
    }
  }

  &__creditCard {
    grid-area: creditCard;
  }

  &__form {
    padding-top: $space-xl;
    padding-right: $space-xl;
    padding-left: $space-xl;
    justify-self: end;
    width: 100%;
    max-width: 540px;
    align-items: flex-end;
    display: flex;
    flex-flow: column;
    @media (max-width: $responsiveValue) {
      padding-top: 200px;
      padding-right: 40px;
      padding-left: 40px;
      max-width: 100%;
    }
  }

  &__steps {
    margin-bottom: 40px;
    @media (max-width: $responsiveValue) {
      display: none;
    }
  }

  &__fields {
    width: 100%;
    display: grid;
    grid-template-columns: 0.95fr 0.05fr 1fr;
    align-items: self-end;
    grid-template-rows: 36px;
    grid-row-gap: 50px;
    grid-column-gap: $space-md;
    grid-template-areas:
      'card_number card_number card_number'
      'card_name card_name card_name'
      'card_date card_date card_cvv'
      'card_installments card_installments card_installments'
      '. send send';
  }
}
</style>
