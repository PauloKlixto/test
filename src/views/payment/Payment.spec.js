/* eslint-disable no-undef */
import { shallowMount } from '@vue/test-utils'
import Payment from './Payment'
import { mockCardType } from '@/views/payment/assets'

describe('Payment', () => {
  const WRAPPER = shallowMount(Payment)

  test('Componente existe', () => {
    expect(WRAPPER.exists()).toBe(true)
  })

  describe('Methods - getCardType', () => {
    test('Quando o card_number está vazio', () => {
      WRAPPER.setData({
        form: { card_number: '' }
      })
      WRAPPER.vm.getCardType()
      expect(WRAPPER.vm.cardType).toEqual(mockCardType)
    })
    test('Quando o card_number não está vazio', () => {
      WRAPPER.setData({
        form: { card_number: '4325' }
      })
      WRAPPER.vm.getCardType()
      expect(WRAPPER.vm.cardType).toEqual({
        company: 'visa',
        name: 'creditCardNumber',
        valid: true
      })
    })
  })

  test('Componente não alterado', () => {
    expect(WRAPPER.html()).toMatchSnapshot()
  })
})
