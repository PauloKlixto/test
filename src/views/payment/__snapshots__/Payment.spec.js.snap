// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment Componente não alterado 1`] = `
"<main class=\\"Payment\\">
  <section class=\\"Payment__sidebar\\">
    <basebackbutton-stub tag=\\"a\\" label=\\"Alterar forma de pagamento\\" isinvertedtheme=\\"true\\" class=\\"Payment__backButton\\"></basebackbutton-stub>
    <basesteps-stub steps=\\"[object Object],[object Object],[object Object]\\" invertcolors=\\"true\\" class=\\"Payment__stepsInSidebar\\"></basesteps-stub>
    <baseheading-stub label=\\"Adicione um novo cartão de crédito\\" icon=\\"credit-card\\" class=\\"Payment__headingText\\"></baseheading-stub>
    <basecreditcard-stub cardnumber=\\"4325\\" cardname=\\"\\" carddate=\\"\\" cardcvv=\\"\\" company=\\"visa\\" class=\\"Payment__creditCard\\"></basecreditcard-stub>
  </section>
  <section class=\\"Payment__form\\">
    <basesteps-stub steps=\\"[object Object],[object Object],[object Object]\\" class=\\"Payment__steps\\"></basesteps-stub>
    <dynamicform-stub model=\\"[object Object]\\" blueprint=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" endpointerrors=\\"[object Object]\\" class=\\"Payment__fields\\"></dynamicform-stub>
  </section>
</main>"
`;
